### PR TITLE
[Compare] - Fix bug when subtitles have different number of paragraphs.

### PR DESCRIPTION
**Pull request by ivandrofly**
Updated

- [x] https://github.com/SubtitleEdit/subtitleedit/commit/f6e917367921d0f459c7b021996d77979e83a8b6#commitcomment-18854926
- [x] https://github.com/SubtitleEdit/subtitleedit/commit/f6e917367921d0f459c7b021996d77979e83a8b6#commitcomment-18857448

Take a look here. @niksedk

### CODE REVIEW

File: src/Forms/Compare.cs [hunk 1]
```diff
@@ -253,19 +249,6 @@ private void CompareSubtitles()
...
-                        break;
-                    }
-                }
-                if (p1 == null)
-                {
-                    sub1.Paragraphs.Insert(index, new Paragraph());
-                }
-                if (p2 == null)
-                {
-                    sub2.Paragraphs.Insert(index, new Paragraph());
```
Comment (ivandrofly) [inline src/Forms/Compare.cs]: This is handled by using minimum subtitle count instead.

File: src/Forms/Compare.cs [hunk 2]
```diff
@@ -210,7 +210,7 @@ private void ButtonOpenSubtitle2Click(object sender, EventArgs e)
                 subtitleListView1.SelectIndexAndEnsureVisible(0);
                 subtitleListView2.SelectIndexAndEnsureVisible(0);
                 labelSubtitle2.Text = openFileDialog1.FileName;
-                if (_subtitle2.Paragraphs.Count > 0)
+                if (_subtitle2.Paragraphs.Count > 0 && _subtitle1?.Paragraphs.Count > 0)
```
Comment (ivandrofly) [inline src/Forms/Compare.cs]: Only do compare when both subtitles contains at least 1 paragraph.

**Comment by niksedk** [comment]
Everything ok now?

**Comment by xylographe** [comment]
@ivandrofly Added minor refact. You can either squash or remove with `git rebase -i`. :)

**Comment by ivandrofly** [comment]
> Everything ok now?

Yeah.

Others fixed have been made but I'll send them in difference PR.

**Pull request merged**